From eb2982ac578982de98c2d425abb17f7f7fc1d75a Mon Sep 17 00:00:00 2001
From: Jeremy McNicoll <jeremy.mcnicoll@windriver.com>
Date: Thu, 20 May 2010 12:07:36 -0400
Subject: [PATCH 06/11] WRHV: Protect set_context

For e500mc set_context uses the PID register to point to
the current process.  Where as the standard e500 needs to
copy the kernel page tables from the init process to the
current process.

Signed-off-by: Jeremy McNicoll <jeremy.mcnicoll@windriver.com>
---
 arch/powerpc/kernel/head_wrhv.S |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/arch/powerpc/kernel/head_wrhv.S b/arch/powerpc/kernel/head_wrhv.S
index d0e9c4a..33ff271 100644
--- a/arch/powerpc/kernel/head_wrhv.S
+++ b/arch/powerpc/kernel/head_wrhv.S
@@ -748,6 +748,7 @@ _GLOBAL(abort)
 	mtspr	SPRN_DBCR0,r13
 	isync
 
+#ifndef CONFIG_WRHV_E500
 _GLOBAL(set_context)
 
 #ifdef CONFIG_BDI_SWITCH
@@ -764,6 +765,7 @@ _GLOBAL(set_context)
 	isync
 #endif
 	blr
+#endif
 
 _GLOBAL(flush_dcache_L1)
 	mfspr	r3,SPRN_L1CFG0
-- 
1.6.5.2

